kratos/linear_solvers/amgcl_ns_solver.h

```diff
@@ -265,6 +265,9 @@ class AMGCL_NS_Solver : public LinearSolver< TSparseSpaceType,
         double resid;
         boost::tie(iters, resid) = solve(rB, rX);
 
+        if (mTol < resid)
```

Comment:
`KRATOS_WARNING_IF`

Reply:
I was thinking, Echo level prints everything for each iteration, so It is hard to recognize whether some iterations have not converged specially in transient simulations. So, this will print the warning without considering the echo level if it is not converged enough only which is really useful to see. This won't add anything if everything is converged.

Reply:
@philbucher I like the idea of printing the warning independent of the echo level. Often, I don't want linear solver information unless something failed to converge. We could used `KRATOS_WARNING_IF` but then we should changed the whole class to use the logger which is probably better in a separate pr.

Reply:
After discussing in person, @sunethwarna please use the correct macro (=> `KRATOS_WARNING_IF`) then I am fine with the changes

Reply:
done :) thanks a lot @philbucher and @msandre